R/FilteredDataset.R

```diff
@@ -314,9 +314,38 @@ FilteredDataset <- R6::R6Class( # nolint
     #' @description
     #' Gets variable names for the filtering.
     #'
+    #' It takes the intersection of the column names
+    #' of the data and `private$.filterable_varnames` if
+    #' `private$.filterable_varnames` has positive length
+    #'
     #' @return (`character` vector) of variable names
     get_filterable_varnames = function() {
-      get_filterable_varnames(self$get_dataset())
+      varnames <- self$get_supported_filter_varnames()
+      if (length(private$.filterable_varnames) > 0) {
+        return(intersect(varnames, private$.filterable_varnames))
```

Comment:
Have you think about preserving the order of filterable varnames? 

```
> intersect(letters, c("c", "b", "a"))
[1] "a" "b" "c"
> intersect(c("c", "b", "a"), letters)
[1] "c" "b" "a"
```

Reply:
I've switch them round so the filtereable_varnames comes first

Reply:
thanks - that makes more sense to me tbh